File: Práctica 2/ArrayQueue.java
public class ArrayQueue {
    private Object[] data;
    private int first;
    private int rear;

    public ArrayQueue(int capacity) {
        data = new Object[capacity];
        first = -1;
        rear = -1;
    }

    public int size() {
        if (isEmpty()) {
            return 0;
        }
        return rear - first + 1;
    }

    public boolean isEmpty() {
        return first == -1 || first > rear;
    }

    public void enqueue(Object e) {
        if (isEmpty()) {
            first = 0;
            rear = 0;
        } else {
            rear++;
        }
        data[rear] = e;
    }

    public Object dequeue() {
        if (isEmpty()) {
            return null;
        }
        Object e = data[first];
        if (first == rear) {
            first = -1;
            rear = -1;
        } else {
            first++;
        }
        return e;
    }

    public Object first() {
        if (isEmpty()) {
            return null;
        }
        return data[first];
    }
}

File: Práctica 3/Estructuras/ArrayQueue.py
class ArrayQueue:
    def __init__(self, capacity):
        self.data = [None] * capacity
        self.first = -1
        self.rear = -1

    def size(self):
        return (len(self.data) - self.first + self.rear) % len(self.data) + 1

    def isEmpty(self):
        return self.size() == 0

    def enqueue(self, e):
        if self.size() < len(self.data):
            self.rear = (self.rear + 1) % len(self.data)
            self.data[self.rear] = e

    def dequeue(self):
        if self.isEmpty():
            return None
        else:
            temp = self.data[self.first]
            self.data[self.first] = None
            self.first = (self.first + 1) % len(self.data)
            return temp

    def First(self):
        if self.isEmpty():
            return None
        return self.data[self.first]
